### PR TITLE
google-cloud-speech: init at 0.30.0

### PR DESCRIPTION
**Pull request by vanschelven**
###### Motivation for this change

Google Cloud Speech API enables developers to convert audio to text;
The provided package is the Python client.

###### Things done

pkgs/development/python-modules/google_cloud_speech/default.nix is the actual goal of the present commit; the other packages are required dependencies.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/google_api_core/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.1.2";
+
+  src = fetchPypi {
+    pname = "google-api-core";
+    inherit version;
+    sha256 = "0qmjswj079w7q7zbnh8p4n2r3f831wymm9hfdlc7zfrini7184xv";
+  };
+
+  propagatedBuildInputs = [ google_auth protobuf googleapis_common_protos requests grpcio ];
+  buildInputs = [ setuptools mock pytest ];
```
Comment (FRidh) [inline pkgs/development/python-modules/google_api_core/default.nix]: checkInputs

File: pkgs/development/python-modules/google_api_core/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  checkPhase = ''
+    py.test
+  '';
+
+  meta = {
+    description = "This library is not meant to stand-alone. Instead it defines common helpers used by all Google API clients.";
+    license = "apache";
```
Comment (FRidh) [inline pkgs/development/python-modules/google_api_core/default.nix]: which specifically? Use `lib.licences`

File: pkgs/development/python-modules/google_api_core/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  };
```
Comment (FRidh) [inline pkgs/development/python-modules/google_api_core/default.nix]: maintainer?

Comment (FRidh) [inline pkgs/development/python-modules/google_api_core/default.nix]: homepage?

File: pkgs/development/python-modules/google_api_core/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, buildPythonPackage, fetchPypi
+, google_auth, protobuf, googleapis_common_protos, requests, grpcio, setuptools, mock, pytest }:
+
+buildPythonPackage rec {
+  name = "google-api-core-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/google_api_core/default.nix]:
```
pname = "google-api-core";
version = "0.1.2";
```
drop `name`

File: pkgs/development/python-modules/google_auth/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, buildPythonPackage, fetchPypi
+, pytest, mock, oauth2client, flask, requests, urllib3, pytest-localserver, six, pyasn1-modules, cachetools, rsa }:
+
+buildPythonPackage rec {
+  name    = "google-auth-${version}";
+  version = "1.2.1";
```
Comment (FRidh) [inline pkgs/development/python-modules/google_auth/default.nix]: same here

File: pkgs/development/python-modules/google_auth/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchPypi {
+    pname = "google-auth";
+    inherit version;
+    sha256 = "041qpwlvpawggasvbfpkx39mkg4dgvivj831x7kinidayrf46w3i";
+  };
+
+  buildInputs = [ pytest mock oauth2client flask requests urllib3 pytest-localserver ];
```
Comment (FRidh) [inline pkgs/development/python-modules/google_auth/default.nix]: checkInputs

File: pkgs/development/python-modules/google_auth/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  # probably never be. We just remove the test to make the tests work again.
+  postPatch = ''rm tests/test__oauth2client.py'';
+
+  checkPhase = ''
+    py.test
+  '';
+
+  meta = {
+    description = "This library simplifies using Google’s various server-to-server authentication mechanisms to access Google APIs.";
+    license = "apache";
```
Comment (FRidh) [inline pkgs/development/python-modules/google_auth/default.nix]: use `lib.licenses`

**Review by FRidh — CHANGES_REQUESTED** [review]
Note the comments I made apply to all expressions.

**Comment by FRidh** [comment]
Also, a commit per expression is needed.

**Comment by vanschelven** [comment]
@FRidh anything I can do to get this PR accepted? I've implemented the changes you requested. I'd rather have it merged now that it still has no conflicts with master.

**Pull request merged**